data-access-layer/eventmesh/ApiServerClient.js

```diff
@@ -429,6 +429,45 @@ class ApiServerClient {
       });
   }
 
+  /**
+   * @description Get Resource in Apiserver with the opts
+   * @param {string} opts.resourceGroup - Unique id of resource
+   * @param {string} opts.resourceType - Name of operation
+   * @param {array} opts.stateList - Array of states of resorces
+   */
+  getResourceListByState(opts) {
+    logger.debug('Get resource list with opts: ', opts);
+    assert.ok(opts.resourceGroup, `Property 'resourceGroup' is required to get resource list`);
+    assert.ok(opts.resourceType, `Property 'resourceType' is required to get resource list`);
+    assert.ok(opts.stateList, `Property 'stateList' is required to fetch resource list`);
+    return Promise.try(() => this.init())
+      .then(() => apiserver.apis[opts.resourceGroup][CONST.APISERVER.API_VERSION]
+        .namespaces(CONST.APISERVER.NAMESPACE)[opts.resourceType].get({
+          qs: {
+            labelSelector: `state in (${_.join(opts.stateList, ',')})`
```

Comment:
This method hardcodes the label 'state'. It will be better if this query itself is taken as input so that it will keep this method generic and one can fetch based on any other label as well. [Look here]( https://github.com/jagadish-kb/service-fabrik-broker/blob/feature/abacus_metering/data-access-layer/eventmesh/ApiServerClient.js#L455-L492) . You can then just rename method to `getResources`